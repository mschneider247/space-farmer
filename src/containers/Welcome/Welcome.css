.div_welcome {
  border: 2px solid grey;
  display: flex;
  flex-direction: column;
  padding: 10%;
  align-items: center;
}

.div_name-input {
  width: 310px;
}

input {
  text-align: center;
  width: 300px;
  margin-bottom: 3%;
}

button {
  width: 305px;
}

input, button {
  height: 40px;
  border-radius: 2%;
}

button:hover {
  background-color: green;
  color: white;
}