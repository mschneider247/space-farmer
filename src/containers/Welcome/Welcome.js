import React, { Component } from 'react';
import './Welcome.css'
import { setUser } from '../../actions';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';

export class Welcome extends Component {
  constructor() {
    super();
    this.state = {
      name: ''
    }
  }

  updateName = event => {
    this.setState({ [event.target.name]: event.target.value})
  }

  handleClick = async (event) => {
    event.stopPropagation();
    await this.props.setUser(this.state.name)
  }

  checkLogin = () => {
    if (this.props.user !== "") {
      return (
        <Redirect to='/proposals'></Redirect>
      )
    }
  }

  render() {
    let reRoute = this.checkLogin();
    return (
      <div className="div_welcome">
        <h1>You are the SPACE <span className="farmer">FARMER</span></h1>
        <div className="div_name-input">
          <input 
            name="name" 
            type="text" 
            placeholder="Enter Your Name"
            onChange={this.updateName}
          />
          <button onClick={(event) => this.handleClick(event)}>Start Farming!</button>
        </div>
        {reRoute}
      </div>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  setUser: userName => dispatch(setUser(userName))
})

export const mapStateToProps = state => ({
  user: state.setUser
})

export default connect(mapStateToProps, mapDispatchToProps)(Welcome)