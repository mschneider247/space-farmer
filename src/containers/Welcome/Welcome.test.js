import React from 'react';
import { shallow } from 'enzyme';
import { Welcome } from './Welcome';

describe('Welcome container', () => {
  it('Welcome should match its snapshot', () => {
    const wrapper = shallow(<Welcome/>)
    expect(wrapper).toMatchSnapshot();
  });
})