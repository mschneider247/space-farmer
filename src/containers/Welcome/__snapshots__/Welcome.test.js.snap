// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome container Welcome should match its snapshot 1`] = `ShallowWrapper {}`;
